pkg/api/v20220201/openshiftcluster.go

```diff
@@ -0,0 +1,277 @@
+
+	// The cluster properties.
+	Properties OpenShiftClusterProperties `json:"properties,omitempty"`
+}
+
+// Tags represents an OpenShift cluster's tags.
+type Tags map[string]string
+
+// OpenShiftClusterProperties represents an OpenShift cluster's properties.
+type OpenShiftClusterProperties struct {
+	// The cluster provisioning state.
+	ProvisioningState ProvisioningState `json:"provisioningState,omitempty"`
+
+	// The cluster profile.
+	ClusterProfile ClusterProfile `json:"clusterProfile,omitempty"`
+
+	// The console profile.
+	ConsoleProfile ConsoleProfile `json:"consoleProfile,omitempty"`
+
+	// The cluster service principal profile.
+	ServicePrincipalProfile ServicePrincipalProfile `json:"servicePrincipalProfile,omitempty"`
+
+	// The cluster network profile.
+	NetworkProfile NetworkProfile `json:"networkProfile,omitempty"`
+
+	// The cluster master profile.
+	MasterProfile MasterProfile `json:"masterProfile,omitempty"`
+
+	// The cluster worker profiles.
+	WorkerProfiles []WorkerProfile `json:"workerProfiles,omitempty"`
+
+	// The cluster API server profile.
+	APIServerProfile APIServerProfile `json:"apiserverProfile,omitempty"`
+
+	// The cluster ingress profiles.
+	IngressProfiles []IngressProfile `json:"ingressProfiles,omitempty"`
+}
+
+// ProvisioningState represents a provisioning state.
+type ProvisioningState string
+
+// ProvisioningState constants.
+const (
+	ProvisioningStateCreating      ProvisioningState = "Creating"
+	ProvisioningStateUpdating      ProvisioningState = "Updating"
+	ProvisioningStateAdminUpdating ProvisioningState = "AdminUpdating"
+	ProvisioningStateDeleting      ProvisioningState = "Deleting"
+	ProvisioningStateSucceeded     ProvisioningState = "Succeeded"
+	ProvisioningStateFailed        ProvisioningState = "Failed"
+)
+
+// ClusterProfile represents a cluster profile.
+type ClusterProfile struct {
+	// The pull secret for the cluster.
+	PullSecret string `json:"pullSecret,omitempty"`
+
+	// The domain for the cluster.
+	Domain string `json:"domain,omitempty"`
+
+	// The version of the cluster.
+	Version string `json:"version,omitempty"`
+
+	// The ID of the cluster resource group.
+	ResourceGroupID string `json:"resourceGroupId,omitempty"`
+}
+
+// ConsoleProfile represents a console profile.
+type ConsoleProfile struct {
+	// The URL to access the cluster console.
+	URL string `json:"url,omitempty"`
+}
+
+// ServicePrincipalProfile represents a service principal profile.
+type ServicePrincipalProfile struct {
+	// The client ID used for the cluster.
+	ClientID string `json:"clientId,omitempty" mutable:"true"`
+
+	// The client secret used for the cluster.
+	ClientSecret string `json:"clientSecret,omitempty" mutable:"true"`
+}
+
+// SoftwareDefinedNetwork constants.
+type SoftwareDefinedNetwork string
+
+const (
+	SoftwareDefinedNetworkOVNKubernetes SoftwareDefinedNetwork = "OVNKubernetes"
+	SoftwareDefinedNetworkOpenShiftSDN  SoftwareDefinedNetwork = "OpenShiftSDN"
```

Comment:
We may not be shipping OVN stable yet, adding this comment here until direction is more clear 😎

Reply:
This is NOT going to be shipped in next stable API.  Please see the following ticket for more details.

https://msazure.visualstudio.com/AzureRedHatOpenShift/_workitems/edit/13110192